src/hooks/use-dropdown-close.js

```diff
@@ -0,0 +1,23 @@
+import { useEffect } from 'react';
+
+const useDropdownClose = (ref, dropdownOpen, handleDropdownClose) => {
+  function handleClickOutside(event) {
+    if ((ref.current && !ref.current.contains(event.target))) {
+      handleDropdownClose();
+    }
+  }
+
+  useEffect(() => {
+    if (dropdownOpen) {
+      document.addEventListener('mousedown', handleClickOutside);
+      document.addEventListener('focusin', handleClickOutside);
+    }
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+      document.removeEventListener('focusin', handleClickOutside);
+    };
+  }, [ref, dropdownOpen]);
```

Comment:
```suggestion
  }, [ref.current, dropdownOpen]);
```

A "ref" is just an object that stores a pointer (i.e. `current`). Theoretically, we want the use effect to _execute_ when the pointer changes -- not the container of the pointer. If we received a new ref but it was still using the same DOM element then we do not care about executing our effect.